src/Aspire.Dashboard/Model/ThemeManager.cs

```diff
@@ -6,24 +6,29 @@
 
 namespace Aspire.Dashboard.Model;
 
-public interface IEffectiveThemeResolver
+public sealed record ThemeSettings(string? SelectedTheme, string EffectiveTheme);
+
+public interface IThemeResolver
 {
-    Task<string> GetEffectiveThemeAsync(CancellationToken cancellationToken);
+    Task<ThemeSettings> GetThemeSettingsAsync(CancellationToken cancellationToken);
 }
 
-public sealed class BrowserEffectiveThemeResolver(IJSRuntime jsRuntime) : IEffectiveThemeResolver, IAsyncDisposable
+public sealed class BrowserThemeResolver(IJSRuntime jsRuntime) : IThemeResolver, IAsyncDisposable
 {
     private readonly IJSRuntime _jsRuntime = jsRuntime;
     private IJSObjectReference? _jsModule;
 
-    public async Task<string> GetEffectiveThemeAsync(CancellationToken cancellationToken)
+    public async Task<ThemeSettings> GetThemeSettingsAsync(CancellationToken cancellationToken)
     {
         if (_jsModule == null)
         {
             _jsModule = await _jsRuntime.InvokeAsync<IJSObjectReference>("import", "/js/app-theme.js").ConfigureAwait(false);
         }
 
-        return await _jsModule.InvokeAsync<string>("getCurrentTheme", cancellationToken).ConfigureAwait(false);
+        var currentTheme = await _jsModule.InvokeAsync<string>("getCurrentTheme", cancellationToken).ConfigureAwait(false);
+        var themeCookieValue = await _jsModule.InvokeAsync<string?>("getThemeCookieValue", cancellationToken).ConfigureAwait(false);
```

Comment:
Given this is a network call, perhaps they should run concurrently rather than serially.

```suggestion
        var currentThemeTask = _jsModule.InvokeAsync<string>("getCurrentTheme", cancellationToken).ConfigureAwait(false);
        var themeCookieValueTask = _jsModule.InvokeAsync<string?>("getThemeCookieValue", cancellationToken).ConfigureAwait(false);
        var currentTheme = await currentThemeTask;
        var themeCookieValue = await themeCookieValueTask;
```